src/librustc_typeck/collect.rs

```diff
@@ -546,16 +552,11 @@ fn convert_field<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
 {
     let tt = ccx.icx(struct_predicates).to_ty(&ExplicitRscope, &field.ty);
     ty_f.fulfill_ty(tt);
-    write_ty_to_tcx(ccx, field.id, tt);
```

Comment:
`tables.node_types` is never used for items now, that's why all `write_ty_to_tcx`s were removed from collect?